mm2src/lp_ordermatch/simple_market_maker.rs

```diff
@@ -0,0 +1,702 @@
+    info!("calculated price is: {}", calculated_price);
+    if cfg.check_last_bidirectional_trade_thresh_hold.unwrap_or(false) {
+        calculated_price = vwap_calculator(calculated_price.clone(), ctx, &cfg).await?;
+    }
+
+    let volume = match cfg.balance_percent {
+        Some(balance_percent) => balance_percent * base_balance.clone(),
+        None => MmNumber::default(),
+    };
+
+    let min_vol: Option<MmNumber> = match cfg.min_volume {
+        Some(min_volume) => {
+            if cfg.max.unwrap_or(false) {
+                Some(min_volume * base_balance.clone())
+            } else {
+                Some(min_volume * volume.clone())
+            }
+        },
+        None => None,
+    };
+    Ok((min_vol, volume, calculated_price))
+}
+
+async fn update_single_order(
+    rates: RateInfos,
+    cfg: SimpleCoinMarketMakerCfg,
+    uuid: Uuid,
+    _order: MakerOrder,
+    key_trade_pair: String,
+    ctx: &MmArc,
+) -> OrderProcessingResult {
+    info!("need to update order: {} of {} - cfg: {}", uuid, key_trade_pair, cfg);
+    let (min_vol, _, calculated_price) = prepare_order(rates, cfg.clone(), key_trade_pair.clone(), ctx).await?;
+
+    let req = MakerOrderUpdateReq {
+        uuid,
+        new_price: Some(calculated_price),
+        max: cfg.max,
+        volume_delta: None,
+        min_volume: min_vol,
+        base_confs: cfg.base_confs,
+        base_nota: cfg.base_nota,
+        rel_confs: cfg.rel_confs,
+        rel_nota: cfg.rel_nota,
+    };
+
+    let resp = match update_maker_order(ctx, req).await {
+        Ok(x) => x,
+        Err(err) => {
+            warn!(
+                "Couldn't update the order {} - for {} - reason: {}",
+                uuid, key_trade_pair, err
+            );
+            return MmError::err(OrderProcessingError::OrderCreationError);
+        },
+    };
+    info!("Successfully update order for {} - uuid: {}", key_trade_pair, resp.uuid);
+    Ok(true)
+}
+
+async fn create_single_order(
+    rates: RateInfos,
+    cfg: SimpleCoinMarketMakerCfg,
+    key_trade_pair: String,
+    ctx: &MmArc,
+) -> OrderProcessingResult {
+    info!("need to create order for: {} - cfg: {}", key_trade_pair, cfg);
+    let (min_vol, volume, calculated_price) = prepare_order(rates, cfg.clone(), key_trade_pair.clone(), ctx).await?;
+
+    let req = SetPriceReq {
+        base: cfg.base.clone(),
+        rel: cfg.rel.clone(),
+        price: calculated_price,
+        max: cfg.max.unwrap_or(false),
+        volume,
+        min_volume: min_vol,
+        cancel_previous: true,
+        base_confs: cfg.base_confs,
+        base_nota: cfg.base_nota,
+        rel_confs: cfg.rel_confs,
+        rel_nota: cfg.rel_nota,
+        save_in_history: true,
+    };
+    let resp = match create_maker_order(ctx, req).await {
+        Ok(x) => x,
+        Err(err) => {
+            warn!("Couldn't place the order for {} - reason: {}", key_trade_pair, err);
```

Comment:
It is a very frequently repeated pattern. Please refactor this using the `LogOnError` trait https://github.com/KomodoPlatform/atomicDEX-API/blob/c663e971119ec78dac493d6f633ea511e969d561/mm2src/common/log.rs#L229 and `?` operator. You may need to add more methods to it or refactor existing ones.

Reply:
Here `create_maker_order` return the created order:

 https://github.com/KomodoPlatform/atomicDEX-API/blob/c663e971119ec78dac493d6f633ea511e969d561/mm2src/common/log.rs#L240
 
 Doesn't return anything.
 
And I need to return an `OrderProcessingError::OrderCreationError` - I've read the usage of those traits, but I do not see why and how to use it respect my initial idea here.

Any hints/tips?

Reply:
> You may need to add more methods to it or refactor existing ones.

I think you can add new methods like 
```rust
fn error/warn_log_passthrough(self, msg: &str) -> Self
``` 
that will meet our requirements.